### PR TITLE
Improve OpenShift::clean reliability

### PR DESCRIPTION
**Pull request by dsimansk**
Related #293

### CODE REVIEW

File: core/src/main/java/cz/xtf/core/openshift/OpenShift.java [hunk 1]
```diff
@@ -967,24 +967,11 @@ public PodShell podShell(Pod pod) {
...
-
-			return false;
-		};
-
-		return new SimpleWaiter(bs, TimeUnit.SECONDS, 20, "Cleaning project.");
+		for (HasMetadata hasMetadata : listRemovableResources()) {
+			log.debug("DELETE :: " + hasMetadata.getKind() + "/" + hasMetadata.getMetadata().getName());
+			resource(hasMetadata).withGracePeriod(0).cascading(true).delete();
+		}
+		return new SimpleWaiter(() -> listRemovableResources().isEmpty(), TimeUnit.SECONDS, 20, "Cleaning project.");
```
Comment (Crumby) [inline core/src/main/java/cz/xtf/core/openshift/OpenShift.java]: I think we can switch back to default one present in [OpenShiftWaiters](https://github.com/xtf-cz/xtf/blob/master/core/src/main/java/cz/xtf/core/openshift/OpenShiftWaiters.java#L60)

Comment (dsimansk) [reply core/src/main/java/cz/xtf/core/openshift/OpenShift.java]: Done

**Comment by mchoma** [comment]
@dsimansk was there any related fix in 4.3.0 client? Can you reference? I dont see how does it relate to #293, we still dont have custom resources in `listRemovableResources` and I dont assume any of standard recources will cascade to delete CR. What am I missing?

**Comment by maschmid** [comment]
I suppose he meant to link to the issue describing the problem with hang -deploy pods on namespace cleanup...

**Comment by mchoma** [comment]
for record: ```fabric8 client sends  {"apiVersion":"v1","kind":"DeleteOptions","gracePeriodSeconds":0,"orphanDependents":true}   when cascading is false
see also https://github.com/fabric8io/kubernetes-client/issues/1614```

**Comment by dsimansk** [comment]
@mchoma Marek is correct. I've mixed the issues together sorry. This PR is aiming purely on stuck resources with orphan finalizers.

**Pull request merged**